Re-enable exactOptionalPropertyTypes once third-party types are fixed

\`exactOptionalPropertyTypes\` was disabled in \`tsconfig/base.json\` to unblock the Vite 8/Rolldown upgrade.

The flag causes type errors in third-party type definitions that don't account for it:
- **Rolldown** (\`MinimalPluginContext\` interface extension conflict)
- **vitest 4.1.0** (\`FailureScreenshotArtifactAttachment.body: undefined\` vs \`string | Uint8Array\`)
- **@types/node** (causes duplicate Vite type resolutions)

melonJS source code itself builds cleanly with the flag enabled — the errors are exclusively in \`node_modules\` type definitions.

### Action
Re-enable \`exactOptionalPropertyTypes: true\` in \`packages/tsconfig/base.json\` once these upstream libraries fix their type definitions. Periodically check by enabling the flag and running \`pnpm build\`.

Upstream type issues are now resolved with current dependency versions (vite 8.0.8, vitest 4.1.4, @types/node 25.6.0).

Enabling `exactOptionalPropertyTypes` now produces only **1 error** in our own code:

```
src/input/gamepad.ts(459,8): error TS2375: Type '{ keyCode: number; value: number; pressed: false; threshold: number | undefined; }' is not assignable to type 'ButtonBinding' with 'exactOptionalPropertyTypes: true'. Consider adding 'undefined' to the types of the target's properties.
```

Ready to re-enable once that `ButtonBinding` type is fixed.

--- a/packages/melonjs/src/input/gamepad.ts
+++ b/packages/melonjs/src/input/gamepad.ts
@@ -20,7 +20,7 @@ interface ButtonBinding {
 	keyCode: number;
 	value: number;
 	pressed: boolean;
-	threshold?: number;
+	threshold?: number | undefined;
 }
 
 interface GamepadBindings {

--- a/packages/tsconfig/base.json
+++ b/packages/tsconfig/base.json
@@ -17,6 +17,7 @@
 		"strictPropertyInitialization": false,
 		"allowImportingTsExtensions": true,
 		"noUnusedParameters": true,
-		"noImplicitOverride": true
+		"noImplicitOverride": true,
+		"exactOptionalPropertyTypes": true
 	}
 }

